Bug in LinuxWebAppSlotV0ToV1

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.5.4

### AzureRM Provider Version

3.95.0

### Affected Resource(s)/Data Source(s)

terraform-provider-azurerm/internal/services/appservice/migration.(*LinuxWebAppSlotV0toV1).UpgradeFunc.LinuxWebAppSlotV0toV1.UpgradeFunc.func

### Terraform Configuration Files

```hcl
Company policies do not allow me to share any files, but the error should be descriptive enough.
```


### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.95.0_x5 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 417 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/migration.(*LinuxWebAppSlotV0toV1).UpgradeFunc.LinuxWebAppSlotV0toV1.UpgradeFunc.func1({0x0?, 0x0?}, 0x0?, {0x0?, 0x0?})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/migration/linux_web_app_slot.go:1742 +0x14c
github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk.StateUpgrades.func1({0x10ad53088, 0x140016f5d70}, 0x0?, {0x109d59700, 0x140002cd200})
        github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk/state_upgrades.go:55 +0x64
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).upgradeJSONState(0x140002f6fd8, {0x10ad53088, 0x140016f5d70}, 0xd?, 0x108d5b092?, 0x14?)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:485 +0x88
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).UpgradeResourceState(0x140002f6fd8, {0x10ad53088?, 0x140016f5c80?}, 0x1400134a040)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:361 +0x384
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).UpgradeResourceState(0x14000e07680, {0x10ad53088?, 0x140016f41b0?}, 0x14001ed40f0)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:757 +0x1f8
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_UpgradeResourceState_Handler({0x10a960800?, 0x14000e07680}, {0x10ad53088, 0x140016f41b0}, 0x14001354000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:395 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002a61e0, {0x10ad7a820, 0x14001a80680}, 0x140037946c0, 0x14001a45bc0, 0x1105fb0d8, 0x0)
        google.golang.org/grpc@v1.58.3/server.go:1374 +0xb44
google.golang.org/grpc.(*Server).handleStream(0x140002a61e0, {0x10ad7a820, 0x14001a80680}, 0x140037946c0, 0x0)
        google.golang.org/grpc@v1.58.3/server.go:1751 +0x7cc
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.58.3/server.go:986 +0xb4
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 71
        google.golang.org/grpc@v1.58.3/server.go:997 +0x160

Error: The terraform-provider-azurerm_v3.95.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2024-03-14T11:35:57.492+0100 [DEBUG] provider: plugin exited
2024-03-14T11:35:57.494+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-03-14T11:35:57.495+0100 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/null/3.2.2/darwin_arm64/terraform-provider-null_v3.2.2_x5 pid=84433
2024-03-14T11:35:57.495+0100 [DEBUG] provider: plugin exited
```


### Expected Behaviour

_No response_

### Actual Behaviour

_No response_

### Steps to Reproduce

Something to do with migrating a slot from .NET 6 to .NET 8.0

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/appservice/migration/linux_web_app_slot.go
+++ b/internal/services/appservice/migration/linux_web_app_slot.go
@@ -1739,8 +1739,8 @@ func (l LinuxWebAppSlotV0toV1) Schema() map[string]*pluginsdk.Schema {
 
 func (l LinuxWebAppSlotV0toV1) UpgradeFunc() pluginsdk.StateUpgraderFunc {
 	return func(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
-		oldId := rawState["service_plan_id"].(string)
-		if oldId == "" {
+		oldId, ok := rawState["service_plan_id"].(string)
+		if !ok || oldId == "" {
 			return rawState, nil
 		}
 		parsedId, err := commonids.ParseAppServicePlanIDInsensitively(oldId)

